lib/chef/resource/openssl_dhparam.rb

```diff
@@ -72,6 +71,15 @@ class OpensslDhparam < Chef::Resource
             end
           end
         end
+        if ::File.exist?(new_resource.path)
```

Comment:
unless dhparam_pem_valid?(new_resource.path)
 # generate key
  # file resource that creates key
else
  # file resource that fixes mode/owner/group
end

Reply:
Could we instead avoid repeating the resource with something like: 
```
        dhparam_contenxt = nil
        unless dhparam_pem_valid?(new_resource.path)
          converge_by("Generating dhparam") do
            dhparam_content = gen_dhparam(new_resource.key_length, new_resource.generator).to_pem
          end
       end
     
        file new_resource.path do
          action :create
          owner new_resource.owner unless new_resource.owner.nil
          group new_resource.group unless new_resource.group.nil?
          mode new_resource.mode
          sensitive true
          content dhparam_content unless dhparam_content.nil?
       end
```